src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-units/attachment-units.component.html

```diff
@@ -109,7 +109,7 @@ <h2 id="page-heading">
         [ngbTooltip]="'artemisApp.attachmentUnit.createAttachmentUnits.processTimeInfo' | artemisTranslate"
         type="button"
         (click)="createAttachmentUnits()"
-        [disabled]="units.length < 1"
+        [disabled]="!units || units!.length < 1"
```

Comment:
I am not quite sure but shouldn't this rather be `units && units.length < 1`. Otherwise, it's just the same condition that decides whether the component is rendered at all -> if it's rendered, it will always be enabled.

Reply:
Actually I just noticed something, if we make units undefined then we can't add new ones as it's not a list anymore, therefore I will not set it if the units are undefined. I think this is a better solution